presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java

```diff
@@ -73,23 +87,37 @@ public void setup()
         server.createCatalog(TEST_CATALOG, "tpch");
         server.installPlugin(new BlackHolePlugin());
         server.createCatalog("blackhole", "blackhole");
-
+        waitForNodeRefresh(server);
         setupTestTables();
     }
 
+    private static void waitForNodeRefresh(TestingPrestoServer server)
+            throws InterruptedException
+    {
+        long start = System.nanoTime();
+        while (server.refreshNodes().getActiveNodes().size() < 1) {
+            Assertions.assertLessThan(nanosSince(start), new Duration(10, SECONDS));
+            MILLISECONDS.sleep(10);
+        }
+    }
+
     private void setupTestTables()
             throws SQLException
     {
         try (Connection connection = createConnection("blackhole", "blackhole");
                 Statement statement = connection.createStatement()) {
             assertEquals(statement.executeUpdate("CREATE TABLE test_table (x bigint)"), 0);
         }
+        executorService = newCachedThreadPool();
     }
 
     @AfterClass
     public void teardown()
+            throws Exception
     {
         closeQuietly(server);
+        executorService.shutdownNow();
+        executorService.awaitTermination(1, MINUTES);
```

Comment:
This isn't needed. We are in the cleanup step, no need to wait for anything to finish.